### ISSUE TITLE
Unity false positive RCS1169

### ISSUE THREAD

**Issue by Benna96** [dotnet/roslynator#1236]
Related to #585, figured better make a new issue rather than necro the old one.

**Product and Version Used**: [Roslynator 2022](https://marketplace.visualstudio.com/items?itemName=josefpihrt.Roslynator2022), version 4.6.1

**Steps to Reproduce**:
- Set the following option:
  ```editorconfig
  roslynator_suppress_unity_script_methods = true
  ```
- Create a private field with the `[SerializeField]` attribute:
  ```csharp
  [SerializeField] private bool foo;
  ```

**Actual Behavior**:
- RCS1169 (Make field read-only.) gets fired. Not that surprising, as according the [docs](https://josefpihrt.github.io/docs/roslynator/configuration/) the option only affects RCS1213.
- It is necessary to explicitly suppress RCS1169.

**Expected Behavior**:
- RCS1169 doesn't get fired, as per the main developer's claim https://github.com/dotnet/roslynator/issues/585#issuecomment-1102694414.
- RCS1169 shouldn't get fired for fields with the `[SerializeField]` attribute, as these fields can be set through Unity's inspector and therefore aren't readonly. `[SerializeField]` doesn't work for readonly fields.

The name of the option doesn't really fit this case in the first place, perhaps introduce a new option along the lines of
```editorconfig
roslynator_suppress_unity_serialized_fields = true|false
```

**Comment by Benna96** [comment]
Agreed, a general umbrella option would be more convenient, can't see why one wouldn't want to suppress both when using Unity.

**Comment by josefpihrt** [comment]
Config option is `roslynator_unity_code_analysis.enabled = true`
